pkg/kubelet/kubelet_node_status_test.go

```diff
@@ -2166,3 +2166,69 @@ func TestNodeStatusHasChanged(t *testing.T) {
 		})
 	}
 }
+
+func TestUpdateNodeAddresses(t *testing.T) {
+	testKubelet := newTestKubelet(t, false /* controllerAttachDetachEnabled */)
+	defer testKubelet.Cleanup()
+	kubelet := testKubelet.kubelet
+	kubeClient := testKubelet.fakeKubeClient
+
+	existingNode := v1.Node{ObjectMeta: metav1.ObjectMeta{Name: testKubeletHostname}}
+	kubeClient.ReactionChain = fake.NewSimpleClientset(&v1.NodeList{Items: []v1.Node{existingNode}}).ReactionChain
+
+	testAddresses := [][]v1.NodeAddress{
```

Comment:
a table test with existing/new address lists would be easier to follow

would like to see tests around:
* nil list -> populated list
* empty list -> populated list
* populated list -> nil list
* populated list -> empty list
* no change with multiple addresses of the same type
* 1 internalIP addresses -> 2 internalIP addresses
* 2 internalIP addresses -> 1 internalIP addresses
* 2 internalIP addresses -> 2 different internalIP addresses

and verify that constructing/applying the patch always results in a set of node addresses semantically equal to the new addresses